pkg/client/transport.go

```diff
@@ -117,6 +117,28 @@ func TLSConfigFor(config *Config) (*tls.Config, error) {
 	return tlsConfig, nil
 }
 
+// tlsConfigKey returns a unique key for tls.Config objects returned from TLSConfigFor
+func tlsConfigKey(config *Config) (string, error) {
+	hasCA := len(config.CAFile) > 0 || len(config.CAData) > 0
+	hasCert := len(config.CertFile) > 0 || len(config.CertData) > 0
+	if hasCA && config.Insecure {
+		return "", fmt.Errorf("specifying a root certificates file with the insecure flag is not allowed")
+	}
+	if err := LoadTLSFiles(config); err != nil {
+		return "", err
+	}
+	switch {
```

Comment:
This makes me nervous. Why not just call LoadTLSFiles, and then hash the entire object? (with the function in util that uses spew.)